aziotctl/src/internal/check/checks/host_connect_iothub.rs

```diff
@@ -68,17 +68,35 @@ impl HostConnectIotHub {
     ) -> Result<CheckResult> {
         use aziot_identityd_config::ProvisioningType;
 
-        let iothub_hostname = match &shared.cfg.iothub_hostname {
-            Some(s) => s,
+        let identityd_config = unwrap_or_skip!(&cache.cfg.identityd);
+
+        let (iothub_hostname, nested) = match &shared.cfg.iothub_hostname {
+            Some(s) => (s, false),
```

Comment:
this raises the question of whether or not we should add a `--gateway-hostname` flag to compliment the existing `--iothub-hostname` flag that already exists.

Obviously not part of this PR, but this is nonetheless something we should think about.